src/main/java/org/apache/commons/text/similarity/LongestCommonSubsequence.java

```diff
@@ -29,36 +29,254 @@
  * </p>
  *
  * <p>
- * This implementation is based on the Longest Commons Substring algorithm
- * from <a href="https://en.wikipedia.org/wiki/Longest_common_subsequence_problem">
- * https://en.wikipedia.org/wiki/Longest_common_subsequence_problem</a>.
+ * As of version 2.0, a more space-efficient of the algorithm is implemented. The new algorithm has linear space
+ * complexity instead of quadratic. However, time complexity is still quadratic in the size of input strings.
  * </p>
  *
- * <p>For further reading see:</p>
+ * <p>
+ * The implementation is based on Hirschberg's Longest Commons Substring algorithm (cited below).
+ * </p>
  *
- * <p>Lothaire, M. <i>Applied combinatorics on words</i>. New York: Cambridge U Press, 2005. <b>12-13</b></p>
+ * <p>For further reading see:</p>
+ * <ul>
+ * <li>
+ * Lothaire, M. <i>Applied combinatorics on words</i>. New York: Cambridge U Press, 2005. <b>12-13</b>
+ * </li>
+ * <li>
+ * D. S. Hirschberg, "A linear space algorithm for computing maximal common subsequences," CACM, 1975, pp. 341--343.
+ * </li>
+ * </ul>
  *
  * @since 1.0
  */
 public class LongestCommonSubsequence implements SimilarityScore<Integer> {
+    /**
+     * A parameter of {@link #algorithmB(int[][], boolean)} and {@link #algorithmC(int, int, int, int)}:
+     * The first input sequence.
+     * We are passing some of the parameters through fields to avoid defining methods with more than 7 parameters
+     * which makes checkstyle plugin to complain.
+     *
+     * @since 2.0
```

Comment:
Not since 2.0. Either leave blank or @since 1.10 (which is the next release according to the `pom.xml` version, which is 1.10.0-SNAPSHOT (i.e. the snapshot part is dropped, and we just cut a new release; eventually there will come a time we will be ready for 2.0, but I don't think we are planning it anytime soon yet)